### PR TITLE
Document and type `simple_functions.py`

### PR DESCRIPTION
**Pull request by tameTNT**
## Overview: What does this pull request change?

- Just some short and sweet documentation for `simple_functions.py` along with a general neatening up. 🙂.
- Small additions with some extra clarity for these functions (same style as my recent PR for `iterables.py` (#2649)

## Links to added or changed documentation pages

- https://manimce--2674.org.readthedocs.build/en/2674/reference/manim.utils.simple_functions.html

## Reviewer Checklist
- [ ] The PR title is descriptive enough for the changelog, and the PR is labeled correctly
- [ ] If applicable: newly added non-private functions and classes have a docstring including a short summary and a PARAMETERS section
- [ ] If applicable: newly added functions and classes are tested

### CODE REVIEW

**Comment by tameTNT** [comment]
Note that the pre-commit hook there changes `Union[int, float]` to `int | float` which is only supported by Python >= 3.10 (Manim should support Python >= 3.7 I believe). I believe PR #2670 is the issue since it only started happening recently:https://github.com/ManimCommunity/manim/pull/2670#issuecomment-1100435612

**Review by behackl — APPROVED** [review]
Thanks for your contribution! I've checked your changes and type hints, I'm happy with all of them. I noticed that some of the one-line summaries were a bit long, so I've pushed a commit with suggestions for shorter versions; and I've changed your examples to doctests to increase our test coverage.

Please cross-review my latest commit; if you are happy I'll merge this.

File: manim/utils/simple_functions.py [hunk 1]
```diff
@@ -108,17 +125,20 @@ def get_parameters(function: Callable) -> MappingProxyType[str, inspect.Paramete
...
+        ('n', 'k')
     """
     return inspect.signature(function).parameters
 
 
 def sigmoid(x: float) -> float:
-    """Returns the output of the logistic function, defined as 1/(1+e^-x)
+    r"""Returns the output of the logistic function.
+    
+    The logistic function, or sigmoid function, is defined
```
Comment (tameTNT) [inline manim/utils/simple_functions.py:140]:
```suggestion
    The logistic function, a common example of a sigmoid function, is defined
```
A minor [technicality on terminology](https://en.wikipedia.org/wiki/Sigmoid_function#:~:text=A%20common%20example%20of%20a%20sigmoid%20function%20is%20the%20logistic%20function) if it matters.

**Review by tameTNT — COMMENTED** [review]
I was wondering while writing the examples whether you could include inline maths - there's my answer! 😂

But this all looks good to me (save the really pedantic terminology suggestion 🙃)

File: manim/utils/simple_functions.py (hunk 1 above)
Comment (behackl) [reply manim/utils/simple_functions.py]: You are perfectly correct!

**Pull request merged**